### PR TITLE
KAFKA-6741:  Disable Selector's idle connection timeout in testNetworkThreadTimeRecorded() test

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
@rajinisivaram   trace logs shows that connection is getting closed due to idle time out. increased the idle connection timeout to 10secs for this test alone. Pls take a look.

```
[2018-04-05 00:21:35,390] TRACE About to close the idle connection from 0 due to being idle for 5170 millis (org.apache.kafka.common.network.Selector:639)
[2018-04-05 00:21:35,393] DEBUG Connection with /127.0.0.1 disconnected (org.apache.kafka.common.network.Selector:527)
java.io.IOException: Broken pipe
	at sun.nio.ch.FileDispatcherImpl.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:47)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at org.apache.kafka.common.network.SslTransportLayer.flush(SslTransportLayer.java:211)
	at org.apache.kafka.common.network.SslTransportLayer.write(SslTransportLayer.java:644)
	at org.apache.kafka.common.network.SslTransportLayer.write(SslTransportLayer.java:677)
	at org.apache.kafka.common.network.SslTransportLayer.write(SslTransportLayer.java:702)
	at org.apache.kafka.common.network.ByteBufferSend.writeTo(ByteBufferSend.java:60)
	at org.apache.kafka.common.network.KafkaChannel.send(KafkaChannel.java:239)
	at org.apache.kafka.common.network.KafkaChannel.write(KafkaChannel.java:210)
	at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:509)
	at org.apache.kafka.common.network.Selector.poll(Selector.java:424)
	at org.apache.kafka.common.network.NioEchoServer.run(NioEchoServer.java:145)
```

File: clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java [hunk 1]
```diff
@@ -635,7 +635,7 @@ public void testApplicationBufferResize() throws Exception {
     @Test
     public void testNetworkThreadTimeRecorded() throws Exception {
         selector.close();
-        this.selector = new Selector(NetworkReceive.UNLIMITED, 5000, new Metrics(), Time.SYSTEM,
+        this.selector = new Selector(NetworkReceive.UNLIMITED, 10000, new Metrics(), Time.SYSTEM,
```
Comment (hachikuji) [inline clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java]: Would it make sense to disable idle timeout with `NO_IDLE_TIMEOUT_MS` for this test? Do we want it to be at play?

Comment (omkreddy) [reply clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java]: Agree, We can set timeout to NO_IDLE_TIMEOUT_MS  for this test.

**Comment by rajinisivaram** [comment]
@omkreddy Is the test failing consistently without this fix and passing consistently with this fix? The test is waiting for a receive and if it fails to receive, then it would hit idle timeout since there is no other traffic, but I am not sure idle timeout is the cause of the failure. There could be some other timing issue that is causing the receive to fail.

**Comment by omkreddy** [comment]
@rajinisivaram Yes, test is failing most of the times without this patch and passing consistently with the fix.  ran 100 times without any error. so i thought, may be system slowness/load is causing the test to fail.

**Review by rajinisivaram — APPROVED** [review]
@omkreddy Thanks for the PR, LGTM.

**Comment by rajinisivaram** [comment]
@omkreddy Thanks for the PR, merging to trunk.

**Pull request merged**